oshi-core/src/main/java/oshi/software/common/AbstractOperatingSystem.java

```diff
@@ -175,6 +175,11 @@ public String toString() {
         sb.append(getManufacturer()).append(' ').append(getFamily()).append(' ').append(getVersion().toString());
         return sb.toString();
     }
+    
+    @Override
+    public OSProcess[] getProcesses(int limit, ProcessSort sort, boolean withOpenFiles) {
```

Comment:
Haha. No, the bits are OK with me. The boolean was super-quick-fix for me to get this running.

Reply:
Well I'll keep thinking.  Properties is more general but requires string based matching and exposes internal implementation.  Bitmask lets us be more in control of defined features...

Reply:
Overloading `getProcesses()` doesn't sit well with me in this case. My main idea is to return an abstraction around the `OSProcess[]` array that can handle selective updates as well as a changing number of entities. A bitmask might be ideal to choose what gets updated, but it should be for internal use only. No user wants to mess with that.

I'll begin updating my [design document](https://github.com/cilki/oshi/blob/4.0.0/oshi-core/DESIGN.md) again after December with a more mature solution.

Reply:
Honestly? I wouldn't worry that much. Let's rather get it done. If your proposal allows for even better speed, I'm for it. Otherwise let's stick to the KISS principle...

Reply:
Given I've already gone down the overloaded road with the limit and sort switches, I'm probably going to go all in with the bitmask implementation for 3.x, at least.   In 4.x if there's a more global way of handling updates using `Properties` objects I may switch to that route.   And 5.x I'll leave to whatever magic you invent. :)